meson.build

```diff
@@ -390,7 +390,8 @@ add_project_link_arguments(link_flags, language: ['c', 'objc'])
 
 # osdep
 cocoa = dependency('appleframeworks', modules: ['Cocoa', 'IOKit', 'QuartzCore'],
-                   required: get_option('cocoa'))
+                   required: get_option('cocoa').require(build_machine.system() == 'darwin',
```

Comment:
There's already a `darwin` bool in the meson.build. You can just use that (ditto for below).